draft-thomson-train-protocol.md

```diff
@@ -401,14 +401,78 @@ Similarly, if there is a strong need to ensure that a rate limit is respected,
 network elements cannot assume that the signaled limit will be respected by
 endpoints.
 
-## Privacy {#privacy}
+# Privacy considerations {#privacy}
 
-Any network element that can observe the content of that packet can read the
-rate limit that was applied.
-
-{:aside}
-> TODO: Write more.
+In our privacy analysis, we studied whether TRAIN could be used by attackers
+to gain information about the ongoing traffic and compromise the privacy
+of clients or servers by exposing their identity of their activity.
 
+Any network element that can observe the content of that packet can read the
+rate limit that was applied.  Any signal is visible on the path, from the point
+at which it is applied to the point at which it is consumed at an endpoint.
+On path elements can also alter the TRAIN signal to try trigger specific
+reactions and gain further knowledge.
+
+In the general case of a client connected to a server through the
+Internet, we believe that TRAIN does not provide much advantage to attackers.
+The identities of the clients and servers are already visible through their
+IP addresses. Traffic analysis tools already provide more information than
+the data rate limits set by TRAIN. However, we are concerned with two avenues
+of attacks: that the passive observation
+of TRAIN might help identify the end points; and that the active manipulation
+of TRAIN signals might help find out the identity of clients or servers
+located behind VPNs or proxies.
```

Comment:
```suggestion
the data rate limits set by TRAIN.

There are two avenues of attack that require more analysis:

* that the passive observation of TRAIN packets might help identify or
  distinguish endpoints; and
* that active manipulation of TRAIN signals might help reveal the
  identity of endpoints that are otherwise hidden behind VPNs or proxies.
```